src/python/turicreate/_deps/__init__.py

```diff
@@ -9,59 +9,122 @@
+        if not self.loaded_:
+            self.loaded_ = True
+            self.module_ = self.init_func_(self.name_)
+        return getattr(self.module_, attr)
+
+    def __str__(self):
+        return "lazy loading of module %s" % self.name_
+
+    def __repr__(self):
+        return "lazy loading of module %s" % self.name_
 
 def __get_version(version):
     # matching 1.6.1, and 1.6.1rc, 1.6.1.dev
     version_regex = '^\d+\.\d+\.\d+'
     version = _re.search(version_regex, str(version)).group(0)
     return _StrictVersion(version)
 
+def __has_module(name):
+    if six.PY2:
+        return imp.find_module(name) is not None
+    else:
+        return importlib.util.find_spec(name) is not None
 
-HAS_PANDAS = True
+HAS_PANDAS = __has_module('pandas')
 PANDAS_MIN_VERSION = '0.13.0'
-try:
-    import pandas
-    if __get_version(pandas.__version__) < _StrictVersion(PANDAS_MIN_VERSION):
-        HAS_PANDAS = False
-        _logging.warn(('Pandas version %s is not supported. Minimum required version: %s. '
-                      'Pandas support will be disabled.')
-                      % (pandas.__version__, PANDAS_MIN_VERSION) )
-except:
-    HAS_PANDAS = False
-    from . import pandas_mock as pandas
-
-
-HAS_NUMPY = True
+
+def _dynamic_load_pandas(name):
+    global HAS_PANDAS
+    if HAS_PANDAS:
+        import pandas
+        if __get_version(pandas.__version__) < _StrictVersion(PANDAS_MIN_VERSION):
+            HAS_PANDAS = False
+            _logging.warn(('Pandas version %s is not supported. Minimum required version: %s. '
+                        'Pandas support will be disabled.')
+                        % (pandas.__version__, PANDAS_MIN_VERSION) )
+    if not HAS_PANDAS:
+        from . import pandas_mock as pandas
+
+    return pandas
+
+pandas = LazyModuleLoader('pandas', _dynamic_load_pandas)
+
+HAS_NUMPY = __has_module('numpy')
 NUMPY_MIN_VERSION = '1.8.0'
-try:
-    import numpy
 
-    if __get_version(numpy.__version__) < _StrictVersion(NUMPY_MIN_VERSION):
-        HAS_NUMPY = False
-        _logging.warn(('Numpy version %s is not supported. Minimum required version: %s. '
-                      'Numpy support will be disabled.')
-                      % (numpy.__version__, NUMPY_MIN_VERSION) )
-except:
-    HAS_NUMPY = False
-    from . import numpy_mock as numpy
+def _dynamic_load_numpy(name):
+    global HAS_NUMPY
+    if HAS_NUMPY:
+        import numpy as _ret
+        if __get_version(_ret.__version__) < _StrictVersion(NUMPY_MIN_VERSION):
+            HAS_NUMPY = False
+            _logging.warn(('Numpy version %s is not supported. Minimum required version: %s. '
+                        'Numpy support will be disabled.')
+                        % (numpy.__version__, NUMPY_MIN_VERSION) )
+    if not HAS_NUMPY:
+        from . import numpy_mock as _ret
+    return _ret
 
+numpy = LazyModuleLoader('numpy', _dynamic_load_numpy)
 
-HAS_SKLEARN = True
+HAS_SKLEARN = __has_module('sklearn')
```

Comment:
So it doesn't look like anything is actually using `HAS_SKLEARN`. So let's just rip this code out completely.

Reply:
removed.